core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -627,7 +627,11 @@ private[chisel3] object Builder {
 
 
   def build[T <: RawModule](f: => T): (Circuit, T) = {
-    dynamicContextVar.withValue(Some(new DynamicContext())) {
+    build(f, new DynamicContext())
+  }
+
+  private [chisel3] def build[T <: RawModule](f: => T, dynamicContext: DynamicContext): (Circuit, T) = {
```

Comment:
Is this the right API, to expose all of dynamicContext, as opposed to perhaps something more targeted like preinitialized / reserved names?

Reply:
So I made the function package private, which should give us the ability to change it in the future. However, I think it actually does make sense to expose the dynamic context, because I've often hit cases with the Instance API that I've been wanting ways to add context-state to a chisel execution.

Reply:
👍